xrootdpyfs/xrdfile.py

```diff
@@ -185,7 +188,14 @@ def read(self, sizehint=-1):
 
         self._assert_mode("r-")
 
-        chunksize = sizehint if sizehint > 0 else self.size
+        chunksize = sizehint if sizehint > 0 else self.size - self._ipp
+
+        if chunksize >= 2147483648:  # 2GB in bytes
+            LOGGER.warning(
```

Comment:
Is the logging necessary here? I mean the library is going to throw an OverFlowError if you pass beyond the limit. Also, as far as I remember, I did some tests, and it's not exactly 2GB the limit for some reason.

Either we just say that chunksizes >X  get set to X so that the read will succeed....alternativelty, we throw an error